// nhsuk-react-components@2.2.2/lib/components/button/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ButtonLink = exports.Button = void 0;
var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _excluded = ["className", "disabled", "secondary", "reverse"],
  _excluded2 = ["className", "role", "draggable", "children", "disabled", "secondary", "reverse"],
  _excluded3 = ["href", "as"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Button = function Button(_ref) {
  var className = _ref.className,
    disabled = _ref.disabled,
    secondary = _ref.secondary,
    reverse = _ref.reverse,
    rest = _objectWithoutProperties(_ref, _excluded);
  return (
    /*#__PURE__*/
    // eslint-disable-next-line react/button-has-type
    _react["default"].createElement("button", _extends({
      className: (0, _classnames["default"])('nhsuk-button', {
        'nhsuk-button--disabled': disabled
      }, {
        'nhsuk-button--secondary': secondary
      }, {
        'nhsuk-button--reverse': reverse
      }, className),
      disabled: disabled,
      "aria-disabled": disabled ? 'true' : 'false'
    }, rest))
  );
};
exports.Button = Button;
Button.defaultProps = {
  type: 'submit'
};
var ButtonLink = function ButtonLink(_ref2) {
  var className = _ref2.className,
    role = _ref2.role,
    draggable = _ref2.draggable,
    children = _ref2.children,
    disabled = _ref2.disabled,
    secondary = _ref2.secondary,
    reverse = _ref2.reverse,
    rest = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/_react["default"].createElement("a", _extends({
    className: (0, _classnames["default"])('nhsuk-button', {
      'nhsuk-button--disabled': disabled
    }, {
      'nhsuk-button--secondary': secondary
    }, {
      'nhsuk-button--reverse': reverse
    }, className),
    role: role,
    "aria-disabled": disabled ? 'true' : 'false',
    draggable: draggable
  }, rest), children);
};
exports.ButtonLink = ButtonLink;
ButtonLink.defaultProps = {
  role: 'button',
  draggable: false
};
var ButtonWrapper = function ButtonWrapper(_ref3) {
  var href = _ref3.href,
    as = _ref3.as,
    rest = _objectWithoutProperties(_ref3, _excluded3);
  if (as === 'a') {
    return /*#__PURE__*/_react["default"].createElement(ButtonLink, _extends({
      href: href
    }, rest));
  }
  if (as === 'button') {
    return /*#__PURE__*/_react["default"].createElement(Button, rest);
  }
  if (href) {
    return /*#__PURE__*/_react["default"].createElement(ButtonLink, _extends({
      href: href
    }, rest));
  }
  return /*#__PURE__*/_react["default"].createElement(Button, rest);
};
var _default = ButtonWrapper;
exports["default"] = _default;
